configuration/uart.md

```diff
@@ -89,6 +89,28 @@ By default, the primary UART is assigned to the Linux console. If you wish to us
 1. At the prompt `Would you like the serial port hardware to be enabled?` answer 'Yes'
 1. Exit raspi-config and reboot the Pi for changes to take effect.
 
+## Enabling early console (earlycon) for Linux
+
+Although the Linux kernel starts the UARTs relatively early in the boot process, it is still long after some critical bits of infrastructure have been set up. A failure in those early stages can be hard to diagnose without access to the kernel log messages from that time. That's the problem that the "earlycon" mechanism was created to work around. Consoles that support earlycon usage present an additional interface to the kernel that allows for simple, synchronous output - printk won't return until the characters have been output to the UART.
+
+Enable earlycon with a kernel command line parameter - add one of the following to `cmdline.txt`, depending on which UART is the primary:
```

Comment:
Is a bit more explanation required regarding how to select the correct `earlycon` entry? Or is this one of those things where any user advanced enough to want to use this feature, will also be advanced enough to figure out which entry they want?

Reply:
The latter, I would say.